File: Anguler/SCM_Anguler/src/app/service/po-line-item.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../environment/environment';
import { POLineItemRequestDTO, POLineItemResponseDTO } from '../component/shared/model/pOLineItemModel';

@Injectable({
  providedIn: 'root',
})
export class PoLineItemService {
  private apiUrl = environment.apiUrl + "po-line-items";

  constructor(private http: HttpClient) { }

  findAll(): Observable<POLineItemResponseDTO[]> {
    return this.http.get<POLineItemResponseDTO[]>(this.apiUrl);
  }

  getById(id: number): Observable<POLineItemResponseDTO> {
    return this.http.get<POLineItemResponseDTO>(`${this.apiUrl}/${id}`);
  }

  save(item: POLineItemRequestDTO): Observable<POLineItemResponseDTO> {
    return this.http.post<POLineItemResponseDTO>(this.apiUrl, item);
  }

  update(id: number, item: POLineItemRequestDTO): Observable<POLineItemResponseDTO> {
    return this.http.put<POLineItemResponseDTO>(`${this.apiUrl}/${id}`, item);
  }

  delete(id: number): Observable<string> {
    return this.http.delete(`${this.apiUrl}/${id}`, { responseType: 'text' });
  }

  trackByNumber(trackingNumber: string): Observable<POLineItemResponseDTO> {
    return this.http.get<POLineItemResponseDTO>(`${this.apiUrl}/track/${trackingNumber}`);
  }
}

File: spring/SCM/src/main/java/com/example/SCM/service/POLineItemService.java
package com.example.SCM.service;



import com.example.SCM.dto.request.POLineItemRequestDTO;
import com.example.SCM.dto.response.POLineItemResponseDTO;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

public interface POLineItemService {

    POLineItemResponseDTO save(POLineItemRequestDTO dto);
    POLineItemResponseDTO update(Long id,POLineItemRequestDTO dto);
    List<POLineItemResponseDTO> findAll();
    Optional<POLineItemResponseDTO> getById(Long id);
    void delete(Long id);
    POLineItemResponseDTO tracking (String trackingNumber);

}
